### DISCUSSION TITLE
Pytorch-lightning is not able to load the model checkpoint

### DISCUSSION THREAD

**Discussion by yashrathi-git** [Lightning-AI/pytorch-lightning#12199] [code help: CV] [answered]
Hi
I first trained the model on kaggle on celeba:
Here's the link to the notebook: https://www.kaggle.com/yashrathikaggle/resnet-gender-detection-with-98-16-accuracy/notebook

While training the trained automatically checkpointed the best val_acc and train_loss. I downloaded the checkpoint on colab and tried to load the model to see the outputs. And it seems the model outputs are random. It doesn't looks like I was able to get the model.

Link to colab: https://colab.research.google.com/drive/1E9eg3BkBQCsyPjmy1TZ3oR-kK7Ti4JXt#scrollTo=waqBK-6WaxrX

Please help if I am doing something wrong here.

**Comment by rohitgr7** [comment, marked as answer]
hey @yashrathi-git !

your code looks correct.
Why do you think that outputs are random?
you can also validate your model after loading using:
```py
model = LitModel.load_from_checkpoint(...)
trainer = Trainer(...)
trainer.validate(model)  # or trainer.test(model)
```
if the above gives you the desired metrics then the model is loaded correctly.

one more thing, you might need to call `model.eval()` after loading the weights from the checkpoint.

**Reply by yashrathi-git** [reply]
hello @rohitgr7 
Yes you're right, the model loads correctly.
The reason I thought it was still random is because it always predicted female with very high confidence. I tried it with some male images(taken from train dataset) and randomly generated images and it outputs female with more than 90% confidence. 

But when I tried to test the model I was able to replicate the results. 

So as you said seems like not using `model.eval()` made the difference. (idx 0 is female) 
![image](https://user-images.githubusercontent.com/57002207/156697729-0229aa4d-e29b-4a18-8fca-082ed57cbfa0.png)
Also in the dataset there are 57% female images.

Thank you!
